component/main.jsonnet

```
@@ -28,19 +30,32 @@ local secrets = {
     DB_VENDOR: params.database.external.vendor,
     DB_ADDR: params.database.external.host,
     DB_PORT: std.toString(params.database.external.port),
+    [if params.database.tls.enabled then 'JDBC_PARAMS']: params.database.tls.jdbcParams,
   },
 };
 
 local db_secret = kube.Secret(params.database.secretname) {
   metadata+: {
     labels+: params.labels,
   },
-  stringData: secrets[params.database.provider],
+  stringData: connection_secrets[params.database.provider],
+};
+
+local db_cert_secret = kube.Secret(params.database.tls.certSecretName) {
+  metadata+: {
+    labels+: params.labels,
+  },
+  type: 'kubernetes.io/tls',
+  stringData: {
+    'tls.key': params.database.tls.serverCertKey,
+    'tls.crt': params.database.tls.serverCert,
+  },
```

If you want to go the route of always having the db cert secret, but change the content depending on the verification mode:

```suggestion
local db_cert_secret = kube.Secret(params.database.tls.certSecretName) {
  metadata+: {
    labels+: params.labels,
  },
  type: 'kubernetes.io/tls',
  stringData: 
    if params.database.tls.verification == 'selfsigned' then
      {
        'tls.key': params.database.tls.serverCertKey,
        'tls.crt': params.database.tls.serverCert,
      }
    else
      {
         'README.txt': 'Keycloak is configured with DB TLS verification mode %s, no custom CA cert required' % [params.database.tls.verification],
      },
```